src/uu/cp/src/cp.rs

```diff
@@ -866,6 +869,44 @@ fn copy_source(
     }
 }
 
+pub fn normalize_path<P: AsRef<Path>>(path: P) -> PathBuf {
+    let mut components = path.as_ref().components().peekable();
+    let mut ret = if let Some(c @ Component::Prefix(..)) = components.peek().cloned() {
+        components.next();
+        PathBuf::from(c.as_os_str())
+    } else {
+        PathBuf::new()
+    };
+
+    for component in components {
+        match component {
+            Component::Prefix(..) => unreachable!(),
+            Component::RootDir => {
+                ret.push(component.as_os_str());
+            }
+            Component::CurDir => {}
+            Component::ParentDir => {
+                ret.pop();
+            }
+            Component::Normal(c) => {
+                ret.push(c);
+            }
+        }
+    }
+    ret
+}
+
+#[cfg(target_os = "windows")]
+fn adjust_canonicalization<P: AsRef<Path>>(p: P) -> PathBuf {
+    const VERBATIM_PREFIX: &str = r#"\\?\"#;
+    let p = p.as_ref().display().to_string();
+    if p.starts_with(VERBATIM_PREFIX) {
+        Path::new(&p[VERBATIM_PREFIX.len()..].to_string()).to_path_buf()
```

Comment:
Rather than manually removing the prefix, you can just check the first path component for `Prefix::Verbatim`, `Prefix::VerbatimUNC`, and `Prefix::VerbatimDisk`.

Reply:
I was about to write out the code to strip with unnecessary allocation, but after thinking about it, why are we stripping the verbatim prefix?  If we truly need to mess with the file paths, wouldn’t it make more sense to add the verbatim prefix to paths that don’t have it given that without the verbatim prefix there’s a fairly low maximum path length?

Reply:
yeah, great idea, this is now done :)